homeassistant/components/openweathermap/config_flow.py

```diff
@@ -108,20 +108,19 @@ def _get_options_schema(self):
         return vol.Schema(
             {
                 vol.Optional(
-                    CONF_MODE,
-                    default=self.config_entry.options.get(
-                        CONF_MODE, DEFAULT_FORECAST_MODE
-                    ),
+                    CONF_MODE, default=self._get_config_value(CONF_MODE)
                 ): vol.In(FORECAST_MODES),
                 vol.Optional(
-                    CONF_LANGUAGE,
-                    default=self.config_entry.options.get(
-                        CONF_LANGUAGE, DEFAULT_LANGUAGE
-                    ),
+                    CONF_LANGUAGE, default=self._get_config_value(CONF_LANGUAGE)
                 ): vol.In(LANGUAGES),
             }
         )
 
+    def _get_config_value(self, key):
+        if self.config_entry.options:
+            return self.config_entry.options[key]
+        return self.config_entry.data[key]
```

Comment:
Please add a test, something like:
```py
assert result["data_schema"]({}) == {
    CONF_MODE: mode_bla_bla,
    CONF_LANGUAGE: language_bla_bla,
}
```

Reply:
Is this not tested [here](https://github.com/home-assistant/core/blob/8bacfcec50cf523a788010e519c6f28d7c770e3e/tests/components/openweathermap/test_config_flow.py#L120) already?

Reply:
Those tests are not testing all the different cases of the default form values.